app/controllers/concerns/template_filter.rb

```diff
@@ -0,0 +1,88 @@
+module TemplateFilter
+  extend ActiveSupport::Concern
+  
+  included do
+    # Applies scoping to the template list
+    def apply_scoping(scope, customizable = false, all = false)
```

Comment:
It is nice to see apply_scoping and calculate_table_scopes refactored outside the templates_controller and templates_model, however a Rails Concern is meant to be as generic as possible and these functionalities are tight to templates only. I would suggest to create a template_filter module under lib like follows:
```
module TemplateFilter
  def self.apply_scoping
  ...
  end
  def self.calculate_table_scopes
  ...
  end
```
and after under models/template.rb
```
include TemplateFilter
```
which that change, you will know that those methods belong to the Template model and are refactored in another file since you will call Template.apply_scoping instead of apply_scoping directly on Paginable::TemplatesController